packages/recoil/recoil_values/Recoil_atom.js

```diff
@@ -292,6 +293,23 @@ function baseAtom<T>(options: BaseAtomOptions<T>): RecoilState<T> {
         return getLoadable(recoilValue).toPromise();
       }
 
+      function observe<S>(
+        recoilValue: RecoilValue<S>,
+        observer: (value: S) => void,
+      ): () => void {
+        const observed = recoilValue;
+        const {release} = store.subscribeToTransactions(store => {
+          const loadable = store
+            .getState()
+            .currentTree?.atomValues.get(recoilValue.key);
+          if (loadable) {
+            const value = loadable.getValue();
+            observer(value);
```

Comment:
When at atom is set to an equivalent (via reference equality) value then it short-circuits the set operation and subscribers such as `onSet()` are not called.  This introduces the ability to subscribe to selectors as well as atoms.    Currently selectors will also only cause subscribing components to re-render if they are changed to a non-reference-equality value and there is a lot of attention paid to this by users.  (Note there are also requests for more fine-grained control over equality (#1416) and skipping downstream node re-evaluation (#1606) as well).  Thought should also be made if something equivalent can be done here.